boards: arm: mimxrt*_evk: boot cleanups

galak opened the pull request:
Enable CONFIG_NXP_IMX_RT_BOOT_HEADER in Kconfig for all mimxrt*_evk boards if we are booting from one of the "external" memories (Qspi, hyperflash)

Also cleaned up the 1064 to always enable the QSPI regardless of what memory we are booting from.

galak committed:
boards: arm: mimxrt10{20,50,60,64}_evk: Enable BOOT HEADER in Kconfig

If we are setting the code to be in HyperFlash (CONFIG_CODE_HYPERFLASH)
or QSPI (CONFIG_CODE_QSPI) we should enable
CONFIG_NXP_IMX_RT_BOOT_HEADER.

Update mimxrt1064_evk to use select like other boards.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
Wasn't sure if QSPI should enable CONFIG_BOOT_FLEXSPI_NAND or is CONFIG_BOOT_FLEXSPI_NOR correct for that?

MaureenHelm commented:
> Wasn't sure if QSPI should enable CONFIG_BOOT_FLEXSPI_NAND or is CONFIG_BOOT_FLEXSPI_NOR correct for that?

NOR is correct.

Can you please make the same change for rt1020 and rt1060 boards too?

galak commented:
> Can you please make the same change for rt1020 and rt1060 boards too?

will do, wanted to make sure 1050 was correct before doing the others.

galak committed:
boards: arm: mimxrt1064_evk: Always have QSPI flash enabled

There's no reason not to have the QSPI flash available even if we aren't
booting from it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@MaureenHelm updated for all evk boards.
